crates/solver/src/liquidity/zeroex.rs

```diff
@@ -107,3 +153,146 @@ impl SettlementHandling<LimitOrder> for OrderSettlementHandler {
         Ok(())
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use shared::zeroex_api::OrderMetadata;
+
+    #[test]
+    fn order_buckets_get_created() {
+        let base_tokens = Arc::new(BaseTokens::new(H160::zero(), &[]));
+        let token_a = H160([0x00; 20]);
+        let token_b = H160([0xff; 20]);
+        let fake_order = [TokenPair::new(token_a, token_b).unwrap()].into_iter();
+        let relevant_pairs = base_tokens.relevant_pairs(fake_order);
+        let order: Order = Default::default();
+        let metadata: OrderMetadata = Default::default();
+        let bogous_order_1 = OrderRecord {
+            order: Order {
+                taker_token: token_a,
+                maker_token: token_b,
+                ..order.clone()
+            },
+            metadata: metadata.clone(),
+        };
+        let bogous_order_2 = OrderRecord {
+            order: Order {
+                taker_token: token_b,
+                maker_token: token_a,
+                ..order
+            },
+            metadata,
+        };
+        let order_buckets = ZeroExLiquidity::generate_order_buckets(
+            [bogous_order_1, bogous_order_2].into_iter(),
+            relevant_pairs,
+        );
```

Comment:
Would be nice to test what happens when you have orders which shouldn't end up in any bucket and a scenario where you have multiple orders ending up in the same bucket.